pkg/providers/amazon/eks/encryption_configuration.go

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eks
+
+import (
+	"strings"
+
+	"emperror.dev/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
+// ValidateEncryptionConfiguration returns an error in case the specified
+// encryption configuration is invalid.
+func ValidateEncryptionConfiguration(encryptionConfig []*eks.EncryptionConfig, clusterLocation string) (err error) {
+	if encryptionConfig == nil ||
```

Comment:
What I was trying to say: `nil != []*eks.EncryptionConfig{}`, so the proper way of checking a slice for empty value is len.

You don't need the extra nil check here though: len takes care of it and will return 0 (if it is a typed nil that has a length): https://play.golang.org/p/8GqjE204U0P

Reply:
Yeah, you are absolutely right, I do know the nil-handling of the `len` function, it just bugs me to hell that a forced-to-have-one-item slice has a singular name and thus I always treated it as an object in my mind and thus missed the simplification chance.

I'm going to fix this to using len.

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/963282696f03dbad5b8db17ab08e5f8e7901a596..5e416a8ca1511e125a88e52e1add1016d5f646b2).